File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.rs
use std::io::{Read, stdin};

fn main() {
    let mut buf = String::new();
    stdin().read_to_string(&mut buf).unwrap();
    buf.split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .filter(|n| *n > 0)
        .map(|n| if is_palindrome(n) { "yes" } else { "no" })
        .for_each(|s| println!("{s}"));
}

fn is_palindrome(number: i32) -> bool {
    let str = number.to_string();
    let rev_str: String = str.chars().rev().collect();

    return str == rev_str;
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.cc
#include <iostream>

using namespace std;

bool IsPalindrome(const string& str) {
  for (int i = 0; i < str.size() / 2; i++) {
    if (str[i] != str[str.size() - 1 - i]) {
      return false;
    }
  }
  return true;
}

int main() {
  while (true) {
    string input;
    cin >> input;
    if (input == "0") {
      break;
    }

    if (IsPalindrome(input)) {
      cout << "yes\n";
    } else {
      cout << "no\n";
    }
  }

  return 0;
}